app/main/views/questionnaire.py

```diff
@@ -127,61 +188,120 @@ def get_thank_you(eq_id, form_type, collection_id):
 
     if not valid:
         return _render_template('household-composition', questionnaire_manager.state, template='questionnaire')
 
-    return _go_to_next_block(current_location='household-composition', eq_id=eq_id,
+    return _go_to_next_block(current_location=this_block, eq_id=eq_id,
                              form_type=form_type, collection_id=collection_id)
 
 
 def _go_to_next_block(current_location, eq_id, form_type, collection_id):
-    navigator = get_navigator(g.schema_json)
-    next_location = navigator.get_next_location(get_answers(current_user), current_location)
+    navigator = Navigator(g.schema_json, get_answer_store(current_user))
+
+    next_location = navigator.get_next_location(current_block_id=current_location)
+
+    if next_location is None:
+        return redirect_to_summary(eq_id, form_type, collection_id)
+
     metadata = get_metadata(current_user)
     logger.info("Redirecting user to next location %s with tx_id=%s", next_location, metadata["tx_id"])
-    return redirect_to_questionnaire_page(eq_id, form_type, collection_id, next_location)
+    return redirect_to_block(eq_id, form_type, collection_id, next_location['group_id'], next_location['group_instance'], next_location['block_id'])
 
 
 def _delete_user_data():
     get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
     session_manager.clear()
 
 
-def redirect_to_questionnaire_page(eq_id, form_type, collection_id, location):
-    return redirect(url_for('questionnaire.get_questionnaire',
+def redirect_to_introduction(eq_id, form_type, collection_id):
+    return redirect(interstitial_url(eq_id, form_type, collection_id, 'introduction'))
+
+
+def redirect_to_summary(eq_id, form_type, collection_id):
+    return redirect(interstitial_url(eq_id, form_type, collection_id, 'summary'))
+
+
+def redirect_to_thank_you(eq_id, form_type, collection_id):
+    return redirect(interstitial_url(eq_id, form_type, collection_id, 'thank-you'))
+
+
+def redirect_to_confirmation(eq_id, form_type, collection_id):
+    return redirect(interstitial_url(eq_id, form_type, collection_id, 'confirmation'))
+
+
+def redirect_to_block(eq_id, form_type, collection_id, group_id, group_instance, block_id):
+    if block_id in ['summary', 'introduction', 'confirmation', 'thank-you']:
+        return redirect(interstitial_url(eq_id, form_type, collection_id, block_id))
+
+    return redirect(url_for('.get_block',
                             eq_id=eq_id,
                             form_type=form_type,
                             collection_id=collection_id,
-                            location=location))
+                            group_id=group_id,
+                            group_instance=group_instance,
+                            block_id=block_id))
+
+
+def interstitial_url(eq_id, form_type, collection_id, block_id):
+    if block_id == 'summary':
```

Comment:
It would require wrangling the block_id, given thank-you url is get_thank_you. I think this is clearer..